internal/otelcollector/config/metric/gateway/config_builder.go

```diff
@@ -86,108 +88,127 @@ func makeServiceConfig() config.Service {
 	}
-	if !*input.Prometheus.Enabled {
+	if !isPrometheusInputEnabled(input) {
 		cfg.Processors.DropIfInputSourcePrometheus = makeDropIfInputSourcePrometheusConfig()
 	}
-	if !*input.Istio.Enabled {
+	if !isIstioInputEnabled(input) {
 		cfg.Processors.DropIfInputSourceIstio = makeDropIfInputSourceIstioConfig()
 	}
-	if !*input.Otlp.Enabled {
+	if input.Otlp != nil && input.Otlp.Disabled {
 		cfg.Processors.DropIfInputSourceOtlp = makeDropIfInputSourceOtlpConfig()
 	}
+}
 
+func declareNamespaceFilters(pipeline *telemetryv1alpha1.MetricPipeline, cfg *Config) {
 	if cfg.Processors.NamespaceFilters == nil {
 		cfg.Processors.NamespaceFilters = make(NamespaceFilters)
 	}
 
-	if shouldFilterByNamespace(input.Runtime.Enabled, input.Runtime.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceRuntime)
-		cfg.Processors.NamespaceFilters[processorName] = makeFilterByNamespaceRuntimeInputConfig(pipeline.Spec.Input.Runtime.Namespaces)
+	input := pipeline.Spec.Input
+	if isRuntimeInputEnabled(input) && shouldFilterByNamespace(input.Runtime.Namespaces) {
+		processorID := makeNamespaceFilterID(pipeline.Name, metric.InputSourceRuntime)
+		cfg.Processors.NamespaceFilters[processorID] = makeFilterByNamespaceRuntimeInputConfig(pipeline.Spec.Input.Runtime.Namespaces)
 	}
-	if shouldFilterByNamespace(input.Prometheus.Enabled, input.Prometheus.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourcePrometheus)
-		cfg.Processors.NamespaceFilters[processorName] = makeFilterByNamespacePrometheusInputConfig(pipeline.Spec.Input.Prometheus.Namespaces)
+	if isPrometheusInputEnabled(input) && shouldFilterByNamespace(input.Prometheus.Namespaces) {
+		processorID := makeNamespaceFilterID(pipeline.Name, metric.InputSourcePrometheus)
+		cfg.Processors.NamespaceFilters[processorID] = makeFilterByNamespacePrometheusInputConfig(pipeline.Spec.Input.Prometheus.Namespaces)
 	}
-	if shouldFilterByNamespace(input.Istio.Enabled, input.Istio.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceIstio)
-		cfg.Processors.NamespaceFilters[processorName] = makeFilterByNamespaceIstioInputConfig(pipeline.Spec.Input.Istio.Namespaces)
+	if isIstioInputEnabled(input) && shouldFilterByNamespace(input.Istio.Namespaces) {
+		processorID := makeNamespaceFilterID(pipeline.Name, metric.InputSourceIstio)
+		cfg.Processors.NamespaceFilters[processorID] = makeFilterByNamespaceIstioInputConfig(pipeline.Spec.Input.Istio.Namespaces)
 	}
-	if shouldFilterByNamespace(input.Otlp.Enabled, input.Otlp.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceOtlp)
-		cfg.Processors.NamespaceFilters[processorName] = makeFilterByNamespaceOtlpInputConfig(pipeline.Spec.Input.Otlp.Namespaces)
+	if input.Otlp != nil && !input.Otlp.Disabled && shouldFilterByNamespace(input.Otlp.Namespaces) {
```

Comment:
After defining the `isOtlpInputDisabled(input telemetryv1alpha1.MetricPipelineInput)` function we can check here if `!isOtlpInputDisabled(input) && shouldFilterByNamespace(input.Otlp.Namespaces)`